cmd/clusterctl/cmd/version_checker.go

```diff
@@ -139,28 +148,43 @@ New clusterctl version available: v%s -> v%s
 
 func (v *versionChecker) getLatestRelease() (*ReleaseInfo, error) {
 	log := logf.Log
+
+	// Try to get latest clusterctl version number from the local state file.
+	// NOTE: local state file is ignored if older than 1d.
```

Comment:
where are we checking for this?

Reply:
it is checked inside readStateFile; if the file is older than 1d, it returns nil, nil